core/src/main/java/org/elasticsearch/action/search/SearchScrollQueryThenFetchAsyncAction.java

```diff
@@ -218,7 +221,15 @@ public void onFailure(Exception t) {
 
     private void finishHim(SearchPhaseController.ReducedQueryPhase queryPhase) {
         try {
-            final InternalSearchResponse internalResponse = searchPhaseController.merge(true, sortedShardDocs, queryPhase, fetchResults);
+            final InternalSearchResponse internalResponse;
+            if(queryPhase.isEmpty())
+            {
+                internalResponse = InternalSearchResponse.empty();
+            }
+            else {
+                internalResponse = searchPhaseController.merge(true, sortedShardDocs, queryPhase, fetchResults);
+            }
+
```

Comment:
Intention was to work with the empty result if there is coming a request for a context that we have released and avoid to going into the fetch phase.